tachyon/zk/air/constraint_system/constraint_system.h

```diff
@@ -0,0 +1,95 @@
+#ifndef TACHYON_ZK_AIR_CONSTRAINT_SYSTEM_CONSTRAINT_SYSTEM_H_
+#define TACHYON_ZK_AIR_CONSTRAINT_SYSTEM_CONSTRAINT_SYSTEM_H_
+
+#include <stddef.h>
+
+#include <memory>
+#include <utility>
+#include <vector>
+
+#include "third_party/eigen3/Eigen/Core"
```

Comment:
This is minor, though. This seems redundant, cause `matrix_types.h` already includes this.

Reply:
Hmm I try to keep transitive includes to minimum following https://google.github.io/styleguide/cppguide.html#Include_What_You_Use

But I agree with you that sometimes it's too verbose. Should I remove?